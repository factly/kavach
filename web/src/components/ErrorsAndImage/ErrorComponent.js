import React from 'react';
import { Result, Button } from 'antd';
import { Link } from 'react-router-dom';

function ErrorComponent({ status, title, link, message }) {
  status = status || 'warning';
  return (
    <Result
      status={status}
      title={title}
      extra={
        <Link to={link}>
          <Button type="primary">{message}</Button>
        </Link>
      }
    />
  );
}

export default ErrorComponent;
