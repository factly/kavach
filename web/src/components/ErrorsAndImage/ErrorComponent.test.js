import React from 'react';
import { shallow } from 'enzyme';

import ErrorComponent from './ErrorComponent';

describe('ErrorComponent', () => {

	it('should render the component', () => {
		const props = { status: 'warning', title: 'Warning', link: '/home', message: 'Go Home' };
		let component = shallow(<ErrorComponent {...props} />);
		expect(component).toMatchSnapshot();
	});

	it('should have link and message as extra', () => {
		const props = { status: 'warning', title: 'Warning', link: '/home', message: 'Go Home' };
		let component = shallow(<ErrorComponent {...props} />);

		expect(component.props().extra.props.to).toEqual('/home');
		expect(component.props().extra.props.children.props.type).toEqual('primary');
	});

	it('should render the component when status undefined', () => {
		const props = { title: 'Warning', link: '/home', message: 'Go Home' };
		let component = shallow(<ErrorComponent {...props} />);
		expect(component).toMatchSnapshot();
		expect(component.props().status).toEqual('warning');
	});

});
