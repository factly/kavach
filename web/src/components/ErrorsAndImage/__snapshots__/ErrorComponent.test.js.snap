// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent should render the component 1`] = `
<Result
  extra={
    <ForwardRef
      to="/home"
    >
      <ForwardRef(InternalButton)
        type="primary"
      >
        Go Home
      </ForwardRef(InternalButton)>
    </ForwardRef>
  }
  status="warning"
  title="Warning"
/>
`;

exports[`ErrorComponent should render the component when status undefined 1`] = `
<Result
  extra={
    <ForwardRef
      to="/home"
    >
      <ForwardRef(InternalButton)
        type="primary"
      >
        Go Home
      </ForwardRef(InternalButton)>
    </ForwardRef>
  }
  status="warning"
  title="Warning"
/>
`;
